src/App.tsx

```diff
@@ -1,18 +1,29 @@
-import React from 'react';
+import React, { useState } from 'react';
 import './App.css';
+import { Pagination } from './components/Pagination';
 import { getNumbers } from './utils';
 
-// eslint-disable-next-line @typescript-eslint/no-unused-vars
-const items = getNumbers(1, 42)
-  .map(n => `Item ${n}`);
-
 export const App: React.FC = () => {
+  const [total] = useState(42);
+  const [perPage, setPerPage] = useState(5);
+  const [currentPage, setCurrentPage] = useState(1);
+
+  const onPageChange = (page: number) => {
+    setCurrentPage(page);
+  };
+
+  const maxEnd = perPage * currentPage;
+  const itemEnd = maxEnd > total ? total : maxEnd;
+  const itemStart = (maxEnd - perPage) + 1;
+
+  const items = getNumbers(itemStart, itemEnd).map(n => `Item ${n}`);
```

Comment:
leave items as a variable outside functions
you have static array of items
manipulate with indexes to know what items to show